module/VuFind/src/VuFind/ILS/Driver/AbstractBase.php

```diff
@@ -66,6 +69,161 @@ public function setConfig($config)
+        if ($nonDefaultFields) {
+            $patron = array_merge($patron, $nonDefaultFields);
+        }
+        // Add cat_username and cat_password after debugging to avoid logging these values into a log file
+        $patron['cat_username'] = $cat_username;
+        $patron['cat_password'] = $cat_password;
+        return array_map(fn ($val) => is_string($val) ? trim($val) : $val, $patron);
+    }
+
+    /**
+     * Create a profile array according to getMyProfile specs defined in the documentation.
+     * Each value is trimmed if they are not null.
+     *
+     * @param ?string $firstname        Profile first name
+     * @param ?string $lastname         Profile last name
+     * @param string  $birthdate        Y-m-d or an empty string
+     * @param ?string $address1         Address 1
+     * @param ?string $address2         Address 2
+     * @param ?string $city             City
+     * @param ?string $country          Country
+     * @param ?string $zip              Postal code
+     * @param ?string $phone            Phone number
+     * @param ?string $mobile_phone     Mobile phone number
+     * @param ?string $expiration_date  Profile expiration date
+     * @param ?string $group            Group i.e. Student, Staff, Faculty, etc
+     * @param ?string $home_library     The locationID value of a pick-up location (see getPickUpLocations)
+     *                                  that should be used as the patron's default
+     * @param array   $nonDefaultFields Non default fields not documented in the documentation.
+     *                                  Merges into the resulting profile array.
+     *
+     * @see https://vufind.org/wiki/development:plugins:ils_drivers#getmyprofile
+     *
+     * @return array
+     */
+    public function createProfileArray(
+        ?string $firstname = null,
+        ?string $lastname = null,
+        ?string $birthdate = null,
+        ?string $address1 = null,
+        ?string $address2 = null,
+        ?string $city = null,
+        ?string $country = null,
+        ?string $zip = null,
+        ?string $phone = null,
+        ?string $mobile_phone = null,
+        ?string $expiration_date = null,
+        ?string $group = null,
+        ?string $home_library = null,
+        array $nonDefaultFields = []
+    ): array {
+        $profile = compact(
+            'firstname',
+            'lastname',
+            'birthdate',
+            'address1',
+            'address2',
+            'city',
+            'country',
+            'zip',
+            'phone',
+            'mobile_phone',
+            'expiration_date',
+            'group',
+            'home_library'
+        );
+        $this->debugDriverResult(__FUNCTION__, $profile, $nonDefaultFields);
+
+        if ($nonDefaultFields) {
+            $profile = array_merge($profile, $nonDefaultFields);
+        }
+        return array_map(fn ($value) => $value ? trim((string)$value) : null, $profile);
```

Comment:
I'd be inclined to leave the null check as it was, just to be sure "falsey" values like `0` don't get converted to null inappropriately. (I don't think it's too likely to happen, but I could imagine something like a group code of 0 in a hypothetical situation).

```suggestion
        return array_map(fn ($value) => null !== $value ? trim((string)$value) : null, $profile);
```